src/main/java/com/backend/blooming/stamp/presentation/dto/response/ReadAllStampResponse.java

```diff
@@ -0,0 +1,41 @@
+package com.backend.blooming.stamp.presentation.dto.response;
+
+import com.backend.blooming.stamp.application.dto.ReadAllStampDto;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public record ReadAllStampResponse(Map<Integer, List<StampInfoResponse>> stamps) {
+
+    public static ReadAllStampResponse from(final ReadAllStampDto readAllStampDto) {
+        final List<StampInfoResponse> stampInfos = readAllStampDto.stamps()
+                                                                  .stream()
+                                                                  .map(StampInfoResponse::from)
+                                                                  .toList();
+        final Map<Integer, List<StampInfoResponse>> stamps =
+                stampInfos.stream()
+                          .collect(Collectors.groupingBy(StampInfoResponse::day));
```

Comment:
아래처럼 stream을 합칠 수 있을 것 같네요!

```suggestion
        final Map<Integer, List<StampInfoResponse>> stamps = 
                readAllStampDto.stamps()
                               .stream()
                               .map(StampInfoResponse::from)
                               .collect(Collectors.groupingBy(StampInfoResponse::day));
```